In [2]:
from mgtwr.sel import SearchMGTWRParameter
from mgtwr.model import MGTWR
import pandas as pd


In [3]:
data = pd.read_csv("example.csv")
coords = data[['longitude', 'latitude']]
t = data[['t']]
X = data[['x1', 'x2']]
y = data[['y']]


In [9]:
coords.shape

(1728, 2)

`bws` tuple description:
1. **`numpy.ndarray`, length 3**: Optimized spatial bandwidths (`opt_bws`).
2. **`numpy.ndarray`, length 3**: Optimized temporal bandwidths (`opt_tau`).
3. **`numpy.ndarray`, length 8**: History of spatial bandwidths over iterations (`np.array(BWs)`).
4. **`numpy.ndarray`, length 8**: History of temporal bandwidths over iterations (`np.array(Taus)`).
5. **`numpy.ndarray`, length 8**: Convergence scores (`np.array(scores)`).
6. **`numpy.ndarray`, length 1728**: Final regression coefficients for each observation and covariate (`Betas`).
7. **`numpy.ndarray`, length 1728**: Final residuals (`err`).
8. **`numpy.float64`**: Initial GTWR spatial bandwidth (`bw_gtwr`).
9. **`numpy.float64`**: Initial GTWR temporal bandwidth (`tau_gtwr`).

In [4]:
sel_multi = SearchMGTWRParameter(coords, t, X, y, kernel='gaussian', fixed=True)
bws = sel_multi.search(multi_bw_min=[0.1], verbose=True, tol_multi=1.0e-4, time_cost=True)

Current iteration: 1 ,SOC: 0.0025955
Bandwidths: 0.7, 0.7, 0.5
taus: 1.2,0.6,0.8
Current iteration: 2 ,SOC: 0.000949
Bandwidths: 0.7, 0.7, 0.5
taus: 1.3,0.4,0.8
Current iteration: 3 ,SOC: 0.0005564
Bandwidths: 0.9, 0.7, 0.5
taus: 2.8,0.4,0.8
Current iteration: 4 ,SOC: 0.0003325
Bandwidths: 0.9, 0.7, 0.5
taus: 3.2,0.2,0.8
Current iteration: 5 ,SOC: 0.0002056
Bandwidths: 0.9, 0.7, 0.5
taus: 3.4,0.2,0.8
Current iteration: 6 ,SOC: 0.0001482
Bandwidths: 0.9, 0.7, 0.5
taus: 3.6,0.2,0.8
Current iteration: 7 ,SOC: 0.0001087
Bandwidths: 0.9, 0.7, 0.5
taus: 3.6,0.2,0.8
Current iteration: 8 ,SOC: 8.22e-05
Bandwidths: 0.9, 0.7, 0.5
taus: 3.6,0.2,0.8
time cost: 0:08:29.896


In [30]:
bws[1]

array([3.6, 0.2, 0.8])